modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/XmlProcessor.java

```
@@ -0,0 +1,915 @@
+                attrName = attrName.toLowerCase(Locale.ROOT);
+            }
+
+            return attrName;
+        }
+    }
+
+    /**
+     * Creates a secure, pre-configured SAX parser factory for XML parsing using XmlUtils.
+     */
+    private static SAXParserFactory createSecureSaxParserFactory() {
+        try {
+            SAXParserFactory factory = XmlUtils.getHardenedSaxParserFactory();
+            factory.setValidating(false);
+            factory.setNamespaceAware(false);
+            return factory;
+        } catch (Exception e) {
+            throw new IllegalStateException("Cannot configure secure XML parsing features", e);
+        }
+    }
+
+    /**
+     * Creates a secure, pre-configured namespace-aware SAX parser factory for XML parsing using XmlUtils.
+     */
+    private static SAXParserFactory createSecureSaxParserFactoryNamespaceAware() {
+        try {
+            SAXParserFactory factory = XmlUtils.getHardenedSaxParserFactory();
+            factory.setValidating(false);
+            factory.setNamespaceAware(true);
+            return factory;
+        } catch (Exception e) {
+            throw new IllegalStateException("Cannot configure secure XML parsing features", e);
+        }
+    }
+
+    /**
+     * Creates a secure, pre-configured SAX parser factory for strict XML parsing using XmlUtils.
+     */
+    private static SAXParserFactory createSecureSaxParserFactoryStrict() {
+        try {
+            SAXParserFactory factory = XmlUtils.getHardenedSaxParserFactory();
+            factory.setValidating(false);
+            factory.setNamespaceAware(false);
+
+            // Try to enable strict parsing features (optional - may not be supported)
+            try {
+                factory.setFeature("http://apache.org/xml/features/validation/check-full-element-content", true);
+            } catch (Exception e) {
+                // Strict parsing features are optional - continue without them if not supported
```

So... If the user has set `strict_parsing`, and it's not supported, is silently ignoring it definitely the right thing to do? I kinda feel it's not.

At the very least, I think we should log.warn the exception and its significance (i.e. that we're going to ignore strict_parsing).

However, might it not be better to actually fail? Then the user can decide whether they'd like to do without `strict_parsing`, or whether they'd like to fix whatever issue with their JDK means it's not supported, or to find some other solution. If you go with my advice from the comment above, I think you can simply omit this inner try-catch block: if, for some reason, the JDK supports the other options but not this one, then the other factories will be initialized and this one will be null and the user will just get an exception if they've set `strict_parsing`. (We should make sure that the exception we throw tells the user that such-and-such a combination of features isn't working, so they can try others.)